### PR TITLE
Use QuerySitePlans in my-sites/plans/main

### PR DESCRIPTION
**Pull request by lamosty**
Fix for #6347.

## Testing Instructions

1. Run Calypso with `make run`;
2. Go to `/plans`; verify everything looks/works as before;
3. Try creating new WordPress site; verify everything looks/works as before;
4. Go to incognito mode and create new account; verify everything looks/works as before, especially the plans step.

cc @gwwar 

Test live: https://calypso.live/?branch=fix/use-query-site-plans-in-plans-main

### CODE REVIEW

File: client/my-sites/plans/main.jsx [hunk 1]
```diff
@@ -184,6 +169,7 @@ const Plans = React.createClass( {
 
 						{ ! hasJpphpBundle && this.showMonthlyPlansLink() }
 						<QueryPlans />
+						<QuerySitePlans />
```
Comment (gwwar) [inline client/my-sites/plans/main.jsx]: We should pass a `siteId` to the query component if it's available.

Comment (gwwar) [inline client/my-sites/plans/main.jsx]: This is working currently, since `SiteNotice` also uses the `QuerySitePlans` component.

Comment (gwwar) [inline client/my-sites/plans/main.jsx]: This is good to 🚢 after you pass through the siteId prop

**Comment by lamosty** [comment]
Thanks for the review @gwwar! Fixed.

**Comment by gwwar** [comment]
:shipit:

**Pull request merged**